cluster/store/meta_class.go

```diff
@@ -121,19 +122,38 @@ func (m *metaClass) CopyShardingState() *sharding.State {
 	return &st
 }
 
-func (m *metaClass) AddProperty(p models.Property) error {
+func (m *metaClass) AddProperty(props ...*models.Property) error {
 	m.Lock()
 	defer m.Unlock()
 
 	// update all at once to prevent race condition with concurrent readers
-	src := m.Class.Properties
-	dest := make([]*models.Property, len(src)+1)
-	copy(dest, src)
-	dest[len(src)] = &p
+	src := filterOutDup(m.Class.Properties, props)
+	dest := make([]*models.Property, len(src)+len(props))
+	copy(dest, append(src, props...))
 	m.Class.Properties = dest
 	return nil
 }
 
+// filterOutDup removes from the old any existing property could cause duplication
+func filterOutDup(old, new []*models.Property) []*models.Property {
```

Comment:
shouldn't this method be a part of `m *metaClass`? why it's not part of metaClass struct?

Reply:
not sure what you mean ? why shall it be part of metClass ?

Reply:
not sure what you mean ? why shall it be part of metaClass ?

Reply:
this `filterOutDup` method, shouldn't it be a method of `metaClass``struct`? doesn't it logically belongs to that `struct`?

Reply:
i don't think so, at least at the moment we don't use but just for this place

Reply:
like we , we won't get any benefits by making it method on  `metaClass` given we don't need the class in any internal impl. of the function

Reply:
so it shouldn't be declared like this `func (m *metaClass) filterOutDuplicates(...)`? bc it might be used in other places, right?

Reply:
I don't think so , do we see benefits for it ?

Reply:
ok, then let's leave it like that. I'm always trying to encapsulate as much as I can into structs, that's why I suggested this change, but I'm fine with leaving it